File: solution/210820/타겟넘버.py
def solution(numbers, target):
    answer = 0
    answer = dfs(numbers, target, 0, 0)
    print(answer)
    return answer

def dfs(numbers, target,  i , result):
    if i== len(numbers):
        if result == target:
            return 1
        else:
            return 0
    else:
       return dfs(numbers, target, i+1, result+numbers[i])+dfs(numbers, target, i+1, result-numbers[i])

File: solution/211001/타겟넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = dfs(numbers, 0, 0, target);
        return answer;
    }
    
    public int dfs(int[] numbers, int current, int depth, int target){
        if (depth == numbers.length){
            if (current == target){
                return 1;
            } else {
                return 0;
            }
        }
        
        return dfs(numbers, current + numbers[depth], depth+1 , target) + dfs(numbers, current - numbers[depth], depth+1 , target);
    }
    
}
